### PR TITLE
Retry logic for 202

### PR DESCRIPTION
**Pull request by WHYTEWYLL**

### CODE REVIEW

File: .github/workflows/dbt_dev_general_weekly.yml [hunk 1]
```diff
@@ -0,0 +1,40 @@
+name: dbt_dev_activity_daily_weekly
+run-name: dbt_dev_activity_daily_weekly
+
+on:
+  workflow_dispatch:
+  schedule:
+    - cron: "10 * * * 6" # 10 minutes past the hour, every hour, on Sunday
```
Comment (eric-laurello) [inline .github/workflows/dbt_dev_general_weekly.yml:7]: this is set to run once on  Saturday: https://crontab.guru/#10_*_*_*_6

**Review by desmond-hui — CHANGES_REQUESTED** [review]

File: .github/workflows/dbt_dev_general_weekly.yml (hunk 1 above)
Comment (WHYTEWYLL) [reply .github/workflows/dbt_dev_general_weekly.yml:7]: The comment is wrong, it should be on Satruday.

File: macros/api_udf/request_repo_data.sql [hunk 2]
```diff
@@ -15,6 +15,16 @@ CREATE TABLE IF NOT EXISTS {{ target.database }}.bronze_api.github_repo_data(
 );
 {% endset %}
 
+{% set event_table %}
+
+CREATE OR REPLACE TABLE {{ target.database }}.bronze_api.log_messages (
```
[the remark this answers is not in the document]
Comment (WHYTEWYLL) [reply macros/api_udf/request_repo_data.sql:20]: Good catch

File: macros/api_udf/request_repo_data.sql [hunk 3]
```diff
@@ -110,17 +145,90 @@ CREATE OR REPLACE PROCEDURE {{ target.database }}.bronze_api.get_github_api_repo
...
+            var count_rows_with_excessive_retries = result_set.getColumnValue(1);
+
+            if (count_rows_with_excessive_retries > 0) {
+                var reset_retries_command = `
+                    UPDATE {{ target.database }}.silver.github_repos
+                    SET retries = 0
+                    WHERE retries > 5
+                `;
+                snowflake.execute({sqlText: reset_retries_command});
+}
```
[the remark this answers is not in the document]
Comment (desmond-hui) [reply macros/api_udf/request_repo_data.sql]: Still not sure of the overall intent of this code. If the goal is to stop retrying endpoints with `>5 retries` in the current run of the stored procedure then the `last_time_queried` value would need to be updated as well. If that is not the goal, then this block is fine as is.

**Review by desmond-hui — APPROVED** [review]

File: macros/api_udf/request_repo_data.sql (hunk 3 above)
[the remark this answers is not in the document]
Comment (WHYTEWYLL) [reply macros/api_udf/request_repo_data.sql]: Nop, thats not the case. Its just a way to send them down on the table. And so we try later with a smaller lambda size.

**Pull request merged**